xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -4675,6 +4675,28 @@ public String getExternalURL(String fullname, String action, String querystring,
     }
 
     /**
+     * Get the full URL of the given {@link DocumentReference}. This also includes the server name of the wiki.
+     *
+     * @param documentReference the document that should be resolved
+     * @param action the action of the URL
+     * @param querystring the URL parameters
+     * @param anchor the anchor of the document
+     * @param context the current XWikiContext
+     * @return the full URL of the given reference
+     *
+     * @since 9.6RC1
+     */
+    public String getExternalURL(DocumentReference documentReference, String action,
```

Comment:
Why not make it more generic with ``getExternalURL(EntityReference, ...)``? We have ``getURL(EntityReference)``.